nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/connect/LauncherRegisterPromptHelper.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * (C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nuxeo
+ */
+
+package org.nuxeo.launcher.connect;
+
+import java.io.Console;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.artofsolving.jodconverter.util.PlatformUtils;
+import org.nuxeo.connect.connector.NuxeoClientInstanceType;
+import org.nuxeo.connect.data.ConnectProject;
+import org.nuxeo.launcher.config.ConfigurationException;
+
+import com.sun.istack.internal.NotNull;
+
+/**
+ * Helper methods to prompt different things to the user if the shell is in interactive mode.
+ *
+ * @author <a href="mailto:ak@nuxeo.com">Arnaud Kervern</a>
+ * @since 8.3
+ */
+public class LauncherRegisterPromptHelper {
+    private static final Log log = LogFactory.getLog(LauncherRegisterPromptHelper.class);
+
+    private static final int PAGE_SIZE = 20;
+
+    public static final String CONNECT_TC_URL = "https://www.nuxeo.com/en/services/nuxeo-trial-tc";
+
+    public static String promptMail() {
+        return prompt("Mail Adress: ", "^.+@.+\\..{2,4}", "Mail is invalid.");
+    }
+
+    public static String promptUsername() {
+        return prompt("Username: ", "^\\w+", "Username cannot be empty.");
+    }
+
+    public static String promptCompany() {
+        return prompt("Company: ", "^\\w+", "Company cannot be empty.");
+    }
+
+    public static String promptProjectName() {
+        return prompt("Project name: ", "^(?:[-\\w]+|)$",
+                "Project name can only contain alphanumeric chars and dashes.");
+    }
+
+    public static String promptDescription() {
+        return prompt("Description: ", "", null);
+    }
+
+    protected static String prompt(String message, String regex, String error) {
+        if (StringUtils.isBlank(regex)) {
+            regex = ".*";
+        }
+
+        Console console = System.console();
+        String value = null;
+        do {
+            if (value != null) {
+                console.printf(error + "\n", value);
+            }
+            value = console.readLine(message);
+        } while (!value.matches(regex));
+
```

Comment:
```
String value = console.readLine(message);
while (!value.matches(regex)) {
    console.printf(error + "\n", value);
    value = console.readLine(message);
}
```

Also, `console.readLine(message)` returns << A string containing the line read from the console, not including any line-termination characters, or **null** if an end of stream has been reached. >>
So you must add `value == null ||` or more likely `value !=null &&` to the condition, depending on the wanted behavior (which must be described in the Javadoc).